source/detail/parser/x86_64.cpp

```diff
@@ -31,6 +32,98 @@ namespace smeagle::x86_64 {
     MEMORY        // Types that will be passed and returned in memory via the stack
   };
 
+  // A RegisterAllocator can provide the next register location
+  class RegisterAllocator {
+  public:
+    RegisterAllocator();
+    std::string getRegisterString(RegisterClass regClass, st::Type *paramType);
+    std::string getRegistersString(std::pair<RegisterClass, RegisterClass> regClasses,
+                                   st::Type *paramType);
+
+  private:
+    std::stack<std::string> intRegisters{{"%r9", "%r8", "%rcx", "%rdx", "%rsi", "%rdi"}};
+    std::stack<std::string> sseRegisters;
+    std::string getNextIntRegister();
+    std::string getNextSseRegister();
+    int nextMultipleEight(int number);
+    int framebase = 8;
+    void updateFramebaseFromType(st::Type *paramType);
+  };
+
+  // Constructor
+  RegisterAllocator::RegisterAllocator() {
+    // Populate the sse register stack
+    for (int i = 1; i < 8; ++i) {
+      sseRegisters.push("xmm" + i);
```

Comment:
"xmm" + i   doesn't concatenate strings- it's pointer arithmetic.

Reply:
ohno

Reply:
This should be fixed. Note to self: c++ is not janky like javascript.

Reply:
I shudder to think of the poor people who have to program in Javascript.